frontend/lib/resolution/can-pass.cpp

```diff
@@ -940,7 +940,6 @@ CanPassResult CanPassResult::canPass(Context* context,
     case QualifiedType::MODULE:
     case QualifiedType::TYPE_QUERY:
     case QualifiedType::INDEX:
-    case QualifiedType::DEFAULT_INTENT:
     case QualifiedType::CONST_INTENT:
```

Comment:
I don't think this is the right way to fix this.

IMO, `DEFAULT_INTENT` and `CONST_INTENT` should  be handled next to each other.

Perhaps, we need some new logic for these. It could consider the type and use the type according to the rules described in https://chapel-lang.org/docs/language/spec/procedures.html#the-const-intent .
 * if the type is sync/single/atomic, the default intent is `ref`, so it require types to match exactly; otherwise the intent is `const ref`
 * if they type is a built-in numeric type <= 64 bits or a complex type <= 128 bits, then the default intent and the const intent are both `const in`

However, we already have `intents.cpp` and `resolveIntent`. The question is, why is `canPass` running on an un-resolved intent?